client/me/form-base/with-form-base.jsx

```diff
@@ -46,7 +46,7 @@ const withFormBase = ( WrappedComponent ) => {
 
 		toggleSetting = ( event ) => {
 			const { name } = event.currentTarget;
-			this.props.setUserSetting( name, ! this.props.userSettings[ name ] );
+			this.props.setUserSetting( name, ! this.getSetting( name ) );
```

Comment:
`this.getSetting` might fall back to an empty string, and while inverting this will result in `false`, it feels like it might be an unexpected effect. Should we tighten up types or cast before we invert it as a boolean or something?

Reply:
Mmh, when I think about this I'd like to defer this until we remove the `withFormBase` hoc. There's a few more things in that form which (as you said in your review comment) need to be ironed out to make the ux a bit smoother. Currently if you change the day for delivery and then reset it the form is still dirty because there's a mismatch between string values the select element returns and integer values which are returned by the API for user settings. Addressing this is not a problem, but I'd like to not do this in the hoc but in the component itself. Would it be okay for you to follow up with those changes in another PR?

Reply:
Sounds good 👍